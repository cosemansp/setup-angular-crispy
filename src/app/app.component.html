<div class="container">
  <h1>
    {{message}}!
  </h1>
  <!-- Sample Font-awesome Icon -->
  <i class="fa fa-thumbs-o-up fa-5x" aria-hidden="true"> </i>
</div>



