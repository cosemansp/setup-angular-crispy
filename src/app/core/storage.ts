export interface Options {
  noPrefix: boolean
}

export class Storage {
  private _prefix: string;
  private _storage: any;

  constructor(storage, prefix = '') {
    this._storage = storage;
    this._prefix = prefix;
  }

  set prefix(prefix: string) {
    this._prefix = prefix;
  }

  get prefix(): string {
    return this._prefix;
  }

  set(key: string, value:  any, options?: Options): void {
    const queryKey = this.getPrefixedKey(key, options);
    this._storage.setItem(queryKey, JSON.stringify(value));
  }

  get<T>(key: string, defaultValue?: T, options?: Options): T {
    const queryKey = this.getPrefixedKey(key, options);
    const value = JSON.parse(this._storage.getItem(queryKey));
    if (value === null) {
      return defaultValue;
    }

    return value;
  }

  rm(key: string) {
    const queryKey = this.getPrefixedKey(key);
    this._storage.removeItem(queryKey);
  }

  keys(): string[] {
    const keys = [];
    const allKeys = Object.keys(this._storage);
    if (this.prefix.length === 0) {
      return allKeys;
    }

    allKeys.forEach((key) => {
      if (key.indexOf(this.prefix) !== -1) {
        keys.push(key.replace(this.prefix, ''));
      }
    });

    return keys;
  }

  flush() {
    if (this.prefix.length) {
      this.keys().forEach((key) => {
        this._storage.removeItem(this.getPrefixedKey(key));
      });
      return;
    };

    this._storage.clear();
  };

  private getPrefixedKey(key, options?: Options) {
    options = options || <Options>{};
    if (options.noPrefix) {
      return key;
    }

    if (!this.prefix) {
      return key;
    }

    return `${this.prefix}.${key}`;
  }
}

export const local = new Storage(localStorage);
export const session = new Storage(sessionStorage);


