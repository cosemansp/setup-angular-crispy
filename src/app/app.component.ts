import { ToastsManager } from 'ng2-toastr/ng2-toastr';
import { Component, ViewContainerRef } from '@angular/core';

import eventAggregator from 'app/services/eventAggregator';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  message = 'App Works';

  constructor(private toastr: ToastsManager, vcr: ViewContainerRef) {
    // setup toaster
    toastr.setRootViewContainerRef(vcr);
    eventAggregator.listen('ERROR').subscribe(error => {
      console.error(error);
      toastr.error(error, 'Oops');
    });
  }
}
