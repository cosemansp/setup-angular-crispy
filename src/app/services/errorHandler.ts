import { Injectable, ErrorHandler } from '@angular/core';

@Injectable()
export class CustomErrorHandler extends ErrorHandler {
  constructor() {
    super(true);
  }

  handleError(error: any) {
    //
    // add custom error handling here
    //

    // delegate to the default handler
    super.handleError(error);
  }
}
