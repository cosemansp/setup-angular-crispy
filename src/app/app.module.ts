/* Angular */
import { NgModule, ErrorHandler } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

/* Modules */
import { HttpModule, Http, XHRBackend, RequestOptions } from '@angular/http';
import { ReactiveFormsModule } from '@angular/forms';
import { ToastModule } from 'ng2-toastr/ng2-toastr';

/* 3th Party dependencies */
import { APP_BASE_HREF } from '@angular/common';
import { ToastOptions } from 'ng2-toastr';
import * as Logger from 'js-logger';

/* Local dependencies */
import { environment } from 'environments/environment';
import { local, session } from './core/storage';
import eventAggregator from './services/eventAggregator';
import { CustomErrorHandler } from './services/errorHandler';

// Components
import { AppComponent } from './app.component';
import { SampleComponent } from './components/sample.component';

// Services
import { CustomHttp } from './services/customHttp';

export function createHttp(backend: XHRBackend, options: RequestOptions) {
  return new CustomHttp(backend, options);
}

@NgModule({
  declarations: [AppComponent, SampleComponent],
  imports: [BrowserModule, HttpModule, ReactiveFormsModule, ToastModule.forRoot()],
  providers: [
    // Custom http
    {
      provide: Http,
      useFactory: createHttp,
      deps: [XHRBackend, RequestOptions],
    },
    // Error handler for appInsights
    {
      provide: ErrorHandler,
      useClass: CustomErrorHandler,
    },
    // Dynamic base-href
    {
      provide: APP_BASE_HREF,
      useValue: '/' + (window.location.pathname.split('/')[1] || ''),
    },
  ],
  bootstrap: [AppComponent],
})
export class AppModule {
  constructor() {
    // setup logging
    const consoleHandler = (<any>Logger).createDefaultHandler();
    Logger.useDefaults();
    Logger.setLevel(environment.logLevel);
    Logger.setHandler(function(messages, context) {
      consoleHandler(messages, context);
      // Add custom handlers here
    });
    const log = Logger.get('app');
    log.info('Started');
    log.info('Config', JSON.stringify(environment, null, '\t'));

    // setup localStorage prefix
    // setup localStorage prefix
    local.prefix = environment.appName;
    session.prefix = local.prefix;
  }
}
