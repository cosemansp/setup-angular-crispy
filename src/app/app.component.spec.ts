import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { ToastsManager } from 'ng2-toastr/ng2-toastr'
import { EventAggregator } from './services/eventAggregator'

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent
      ],
      providers: [
        EventAggregator,
        // dummy ToastManager
        {
          provide: ToastsManager, useFactory: () => {
            return { setRootViewContainerRef() {} };
          },
        }
      ]
    }).compileComponents();
  }));

  it('should compile successfully', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
