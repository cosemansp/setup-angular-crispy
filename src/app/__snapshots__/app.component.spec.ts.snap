// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should compile successfully 1`] = `
<app-root
  eventAggregator={[Function EventAggregator]}
  message={[Function String]}
  toastr={[Function Object]}
>
  <div
    class="container"
  >
    <h1>
      App Works!
    </h1>
    <i
      aria-hidden="true"
      class="fa fa-thumbs-o-up fa-5x"
    >
      
    </i>
  </div>
</app-root>
`;
